app/server/src/db/userStore.ts

```diff
@@ -61,64 +81,75 @@ export class UserStore {
-            const samplesCount = await this.getProjectSampleCount(projectId);
-            result.push({
-                id: projectId,
-                name: values[0],
-                hash: values[1],
-                timestamp: parseInt(values[2]),
-                samplesCount
-            });
+            const projectData = await this._redis.hgetall(this._projectKey(projectId));
+            if (!projectData.deleted_at) {
+                const samplesCount = await this.getProjectSampleCount(projectId);
+
+                result.push({
+                    id: projectId,
+                    name: projectData.name,
+                    hash: projectData.hash,
+                    timestamp: parseInt(projectData.timestamp),
+                    samplesCount
+                });
+            }
         }));
 
         return result;
     }
 
     async saveSketch(projectId: string, sampleHash: string, filename: string, sketch: Record<string,unknown> ) {
+        await this._validProject(projectId);
         const sampleId = this._sampleId(sampleHash, filename);
         await this._redis.sadd(this._projectSamplesKey(projectId), sampleId);
         await this._redis.hset(this._projectSampleKey(projectId, sampleId), "sketch", JSON.stringify(sketch));
     }
 
     async saveAMR(projectId: string, sampleHash: string, amr: AMR) {
+        await this._validProject(projectId);
         const sampleId = this._sampleId(sampleHash, amr.filename);
         await this._redis.sadd(this._projectSamplesKey(projectId), sampleId);
         await this._redis.hset(this._projectSampleKey(projectId, sampleId), "amr", JSON.stringify(amr));
     }
-    
+
     async getSketch(projectId: string, sampleHash: string, filename: string): Promise<Record<string, unknown>> {
+        await this._validProject(projectId);
         const sampleId = this._sampleId(sampleHash, filename);
         const sketchString = await this._redis.hget(this._projectSampleKey(projectId, sampleId), "sketch");
         return JSON.parse(sketchString);
     }
 
     async getAMR(projectId: string, sampleHash: string, fileName: string): Promise<AMR> {
+        await this._validProject(projectId);
         const sampleId = this._sampleId(sampleHash, fileName);
         const amrString = await this._redis.hget(this._projectSampleKey(projectId, sampleId), "amr");
         return JSON.parse(amrString);
     }
     async deleteSample(projectId: string, sampleHash: string, filename: string) {
+        await this._validProject(projectId);
         const sampleId = this._sampleId(sampleHash, filename);
         await this._redis.srem(this._projectSamplesKey(projectId), sampleId);
         await this._redis.del(this._projectSampleKey(projectId, sampleId));
     }
 
     async getSample(projectId: string, sampleHash: string, filename: string): Promise<{ sketch: Record<string, unknown>; amr: AMR }> {
+        await this._validProject(projectId);
```

Comment:
This gets repetitive/cluttery. Does anyone know if there is a way to specify "run this util before each function in the following list" at the top of the file or something?

Reply:
yeah thats what i was thinking too but nothing nice honestly I could think of

Reply:
Maybe you could use set up separate middleware that runs before the main route handler, which checks if there is a projectId in request parameters and if so validates it?

Reply:
I haven't done this, let me know if you think it's blocking

Reply:
No, just a thought!